monitoring/opencensus/main_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func TestStartingServer(t *testing.T) {
+	// tc := testutil.SystemTest(t)
+	// test build
+	m := testutil.BuildMain(t)
+	if !m.Built() {
+		t.Fatalf("failed to build app")
+	}
+
+	// test main endpoint
+	req, err := http.NewRequest("GET", "/", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(handle)
+	handler.ServeHTTP(rr, req)
```

Comment:
This should run the binary built by `BuildMain` directly -- the purpose of that is to test `package main` samples fully.

Reply:
I think I might need your help - how would I do that?

Reply:
Thinking about it more, it's a little tricky to run a binary that listens for HTTP requests.

Instead, there is no need for `handler` and `handler.ServeHTTP`. You can just call the handler directly, like this:

https://github.com/GoogleCloudPlatform/golang-samples/blob/ac1f006385cdadc3bccbbfb3fa69b12e3990e077/functions/helloworld/hello_http_test.go#L35-L43

Reply:
Got it - thank you, changed.